source/CScriptEngine.cpp

```diff
@@ -662,12 +671,51 @@ namespace CLEO
         inj.InjectFunction(&opcode_004E_hook, gvm.TranslateMemoryAddress(MA_OPCODE_004E));
     }
 
+    CScriptEngine::CScriptEngine()
+    {
+        CustomMission = nullptr;
+    }
+
+    CScriptEngine::~CScriptEngine()
+    {
+        TRACE("Unloading scripts...");
+        RemoveAllCustomScripts();
+    }
+
     CleoSafeHeader safe_header;
     ThreadSavingInfo *safe_info;
     unsigned long *stopped_info;
     std::unique_ptr<ThreadSavingInfo[]> safe_info_utilizer;
     std::unique_ptr<unsigned long[]> stopped_info_utilizer;
 
+    void CScriptEngine::Initialize()
+    {
+        unsigned char missionPackIdx = *((unsigned char*)0xB72910); // TODO: use CGame::bMissionPackGame instead
+
+        if (missionPackIdx == 0)
+        {
+            //MainScriptFileDir += "0:\\data\\script"; // at game root TODO: enable when CLEO virtual paths available
+            MainScriptFileDir = std::string(MAX_PATH, '\0');
+            GetModuleFileNameA(NULL, MainScriptFileDir.data(), MAX_PATH);
+            MainScriptFileDir.resize(strlen(MainScriptFileDir.data()));
+            MainScriptFileDir = std::filesystem::path(MainScriptFileDir).parent_path().string(); // remove executable name
+            MainScriptFileDir += "\\data\\script";
+            
+            MainScriptFileName = "main.scm";
+        }
+        else
+        {
+            //MainScriptFileDir = "1:\\MPACK\\MPACK"; // at user data TODO: enable when CLEO virtual paths available 
+            MainScriptFileDir = std::string(MAX_PATH, '\0');
+            SHGetFolderPathA(NULL, CSIDL_PERSONAL, NULL, SHGFP_TYPE_CURRENT, MainScriptFileDir.data());
+            MainScriptFileDir.resize(strlen(MainScriptFileDir.data()));
+            MainScriptFileDir += "\\GTA San Andreas User Files";

```

Comment:
.data:00C92368   gta_user_dir_path db 100h dup(?)

absolute path to user files directory

Reply:
Done